### PR TITLE
Error window level

### PR DESCRIPTION
**Pull request by omarZrayka23**
change request inn plant get and windowError

### CODE REVIEW

File: src/providers/RequestsContext.js [hunk 1]
```diff
@@ -7,12 +7,13 @@ import jwt from 'jwt-decode'
 
 import { AuthContext } from 'src/providers/AuthContext'
 import ErrorWindow from 'src/components/Shared/ErrorWindow'
+import Error from 'src/components/Shared/Error'
```
Comment (hadichahine) [inline src/providers/RequestsContext.js:10]: Why do I have an Error?

Comment (hadichahine) [inline src/providers/RequestsContext.js:10]: Why do I have an Error?

File: src/pages/plants/index.js
Comment (hadichahine) [inline src/pages/plants/index.js:1]: What did we change here?

File: src/pages/plants/Forms/PlantForm.js
Comment (hadichahine) [inline src/pages/plants/Forms/PlantForm.js:1]: and here?

File: src/pages/clients-list/index.js [hunk 2]
```diff
@@ -11,19 +11,17 @@ import { formatDateDefault } from 'src/lib/date-helper'
...
 
 const ClientsList = () => {
   const { stack } = useWindow()
-
-  //control
   const { getRequest } = useContext(RequestsContext)
-
-  //error
   const [errorMessage, setErrorMessage] = useState(null)
+  const { stack: stackError } = useError()
```
Comment (hadichahine) [inline src/pages/clients-list/index.js:24]: Why do we have useError and Error used in this one?

File: src/pages/_app.js [hunk 3]
```diff
@@ -142,13 +142,19 @@ const App = props => {
                               <AclGuard aclAbilities={aclAbilities} guestGuard={guestGuard} authGuard={authGuard}>
                                 <PrimeReactProvider>
                                   {getLayout(
-                                    <ErrorProvider key={typeof window !== 'undefined' ? window.location.pathname : ''}>
-                                      <WindowProvider
+                                    <RequestsProvider
+                                      key={typeof window !== 'undefined' ? window.location.pathname : ''}
+                                    >
+                                      <ErrorProvider
```
Comment (hadichahine) [inline src/pages/_app.js:148]: Make sure CashInvoice still works with this (uses useError)

File: src/components/Shared/Table.js
Comment (hadichahine) [inline src/components/Shared/Table.js:1]: what changed here?

File: src/components/Shared/DeleteDialog.js
Comment (hadichahine) [inline src/components/Shared/DeleteDialog.js:1]: changes here?

File: src/components/ConfirmationDialog.js
Comment (hadichahine) [inline src/components/ConfirmationDialog.js:1]: Any changes here?

File: src/components/ConfirmationDialog.js [hunk 4]
```diff
@@ -1,41 +1,69 @@
...
+  openCondition,
+  closeCondition,
+  DialogText,
+  okButtonAction,
+  cancelButtonAction,
+  fullScreen = true
+}) => {
+  return !fullScreen ? (
+    openCondition && (
+      <Window Title='Confirmation' width={450} height={120} onClose={closeCondition}>
```
Comment (hadichahine) [inline src/components/ConfirmationDialog.js:15]: Why is window being used directly?

File: src/providers/RequestsContext.js (hunk 1 above)
Comment (omarZrayka23) [reply src/providers/RequestsContext.js:8]: new window error

Comment (omarZrayka23) [reply src/providers/RequestsContext.js:8]: new window error

File: src/components/Shared/Table.js
Comment (omarZrayka23) [reply src/components/Shared/Table.js:1]: dialog confirm where delete row

File: src/components/Shared/DeleteDialog.js
Comment (omarZrayka23) [reply src/components/Shared/DeleteDialog.js:1]: dialog confirm where delete row

File: src/components/ConfirmationDialog.js (hunk 4 above)
Comment (omarZrayka23) [reply src/components/ConfirmationDialog.js:15]: dialog

File: src/pages/clients-list/index.js (hunk 2 above)
Comment (omarZrayka23) [reply src/pages/clients-list/index.js:24]: We will addressed the issue during the refactoring phase, but now the error window includes screen page only not all window.

File: src/components/Shared/Window.js [hunk 5]
```diff
@@ -138,15 +132,17 @@ return (
...
-                <IconButton
-                  tabIndex={-1}
-                  edge='end'
-                  onClick={() => setExpanded(!expanded)}
-                  data-is-expanded={expanded}
-                  aria-label='expand'
-                >
-                  <OpenInFullIcon /> {/* Add the icon for expanding */}
-                </IconButton>
+                {fullScreen && (
```
Comment (hadichahine) [inline src/components/Shared/Window.js:135]: Let's name it canExpand

File: src/error/index.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import React, { useContext, useState } from 'react'
-import ErrorWindow from 'src/components/Shared/ErrorWindow'
+import Error from 'src/components/Shared/PageError'
```
Comment (hadichahine) [inline src/error/index.js:2]: PageError

File: src/providers/RequestsContext.js [hunk 7]
```diff
@@ -179,11 +185,12 @@ const RequestsProvider = ({ children }) => {
...
 
-  return <RequestsContext.Provider value={values}>{children}
-  {error && (
-    <ErrorWindow open={true} onClose={()=>setError(false)}message={error} />
-  )}
-</RequestsContext.Provider>
+  return (
+    <>
+      <RequestsContext.Provider value={values}>{children}</RequestsContext.Provider>
+      {error && <Error height={400} open={true} onClose={() => setError(false)} message={error} />}
```
Comment (hadichahine) [inline src/providers/RequestsContext.js:191]: PageError

File: src/pages/clients-list/index.js [hunk 8]
```diff
@@ -196,7 +190,7 @@ const ClientsList = () => {
           paginationType='client'
         />
         {errorMessage?.error && (
-          <ErrorWindow open={errorMessage} onClose={() => setErrorMessage(null)} message={errorMessage} />
+          <Error open={errorMessage} height={100} onClose={() => setErrorMessage(null)} message={errorMessage} />
```
Comment (hadichahine) [inline src/pages/clients-list/index.js:193]: PageError

Comment (hadichahine) [inline src/pages/clients-list/index.js:193]: Do we need height?

File: src/providers/RequestsContext.js [hunk 9]
```diff
@@ -80,10 +84,12 @@ const RequestsProvider = ({ children }) => {
         LanguageId: user.languageId
       },
       data: bodyFormData
-    }).then(res => res.data).catch(error => {
-      setError(error); // Set the error state
-      throw error;
     })
+      .then(res => res.data)
+      .catch(error => {
+        setError(error) // Set the error state
```
Comment (hadichahine) [inline src/providers/RequestsContext.js:90]: Can we use stack here?

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/providers/RequestsContext.js [hunk 10]
```diff
@@ -4,15 +4,14 @@ import { createContext, useContext, useState } from 'react'
 // ** 3rd Party Imports
 import axios from 'axios'
 import jwt from 'jwt-decode'
-
 import { AuthContext } from 'src/providers/AuthContext'
-import ErrorWindow from 'src/components/Shared/ErrorWindow'
+import ErrorPage from 'src/components/Shared/PageError'
```
Comment (hadichahine) [inline src/providers/RequestsContext.js:8]: WHy do we need this? Should use useError

File: src/components/ConfirmationDialog.js [hunk 11]
```diff
@@ -1,41 +1,69 @@
...
-            maxWidth="xs"
+const ConfirmationDialog = ({
+  openCondition,
+  closeCondition,
+  DialogText,
+  okButtonAction,
+  cancelButtonAction,
+  fullScreen = true
+}) => {
+  return !fullScreen ? ( //dialog full screen or page
```
Comment (hadichahine) [inline src/components/ConfirmationDialog.js:13]: This is still showing fullscreen?

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/providers/CommonContext.js [hunk 12]
```diff
@@ -8,59 +8,59 @@ import ErrorWindow from 'src/components/Shared/ErrorWindow'
...
-        })
-    }
-    
-    const values = {
-      fillDocumentTypeStore,
-      getAllKvsByDataset
-    }
+      .catch(error => {
+        stackError({ message: error, height: 400 })
+        throw error
```
Comment (hadichahine) [inline src/providers/CommonContext.js:50]: Check if throw is needed.

**Review by hadichahine — CHANGES_REQUESTED** [review]

**Review by hadichahine — APPROVED** [review]

**Pull request merged**